library/common/config/config.cc

```diff
@@ -206,6 +209,8 @@ const char* config_template = R"(
     typed_config: *local_error_config
   - name: envoy.filters.http.dynamic_forward_proxy
     typed_config: *dfp_config
+  - name: envoy.filters.http.socket_tag
+    typed_config: *socket_tag_config
```

Comment:
If we're going to have this on by default, we should call out in the release notes

Reply:
Also I wonder if it should live in library/common/extensions/filters/http/network_configuration/filter.cc ?
That's where @Augustyniak is doing the proxy work I believe

Reply:
Oh, interesting. So instead of a new filter, we'd just add this logic to the existing network_configuration filter? That's a reasonable idea. @Augustyniak, what do you think?

Reply:
@RyanTheOptimist  I do not have a strong opinion here but if I had to choose I think that your current approach is the one I would go with.

I think that in the case of my [proxy PR](https://github.com/envoyproxy/envoy-mobile/pull/2416) there is an extra argument to be made for keeping the proxy logic as part of the existing `network_configuration/filter` since my filter proxy logic retrieves the current proxy settings from the `connectivity_manager` to late apply it to the stream (`streamInfo()->filterState()->setData(...)`) and in general our `network_configuration/filter` and `connectivity_manager` seem to be somehow 'paired'. So if I add something to `connectivity_manager` and I need a corresponding filter logic I put that filter logic in `network_configuration/filter` filter.

Since your logic does not depend on the `connectivity_manager` at all I think that I would keep it separate from our connectivity filter but then again - it's not a strong opinion and I could be persuaded that either approach is the better one.